policies/gitlab/enterprise.rego

```diff
@@ -0,0 +1,225 @@
+	count(public) > 0
+}
+
+# METADATA
+# scope: rule
+# title: Default Group Visibility Should Not Be Public
+# description: >
+#     Your server is configured by default that every new group is created public. This means that it will be accessible by any member and unauthenticated user. It is recommended to turn the default option to be private
+#     and let admins control public visibility explicitly when needed.
+# custom:
+#   severity: HIGH
+#   remediationSteps:
+#     - Go to the admin page: Menu -> Admin
+#     - Press Settings -> General
+#     - Expand "Visibility and access controls" section
+#     - Under "Default group visibility" toggle "Private"
+#     - Press "Save Changes"
+default default_group_visibility_is_public = true
+
+default_group_visibility_is_public = false {
+	not input.default_group_visibility == "public"
+}
+
+# METADATA
+# scope: rule
+# title: Default Project Visibility Should Not Be Public
+# description: >
+#     Your server is configured by default that every new project is created public. This means that it will be accessible by any member and unauthenticated user. It is recommended to turn the default option to be private
+#      and let admins control public visibility explicitly when needed.
+# custom:
+#   severity: HIGH
+#   remediationSteps:
+#     - Go to the admin page: Menu -> Admin
+#     - Press Settings -> General
+#     - Expand "Visibility and access controls" section
+#     - Under "Default project visibility" toggle "Private"
+#     - Press "Save Changes"
+default default_project_visibility_is_public = true
+
+default_project_visibility_is_public = false {
+	not input.default_project_visibility == "public"
+}
+
+# METADATA
+# scope: rule
+# title: Sign-up Confirmation Email Should Be Mandatory
+# description: >
+#     This security option enforces any new user to confirm their email address, and ensures the user is verified. This increases the security and integrity of the identities created within the system, and prevents situations
+#     where a user is spoofed by a malicious actor while using a legitimate corporate email address.
+# custom:
+#   severity: MEDIUM
+#   remediationSteps:
+#     - Go to the admin page: Menu -> Admin
+#     - Press Settings -> General
+#     - Expand "Sign-up restrictions" section
+#     - Toggle "Send confirmation email on sign-up"
+#     - Press "Save Changes"
+default send_user_confirmation_email_not_enabled = true
+
+send_user_confirmation_email_not_enabled = false {
+	input.send_user_confirmation_email
+}
+
+# METADATA
+# scope: rule
+# title: Unauthenticated Requests Rate Limit Should Be Enabled
+# description: >
+#     The servers allows restricting the limit of unauthenticated requests. It is recommended to turn it on as a security and reliability measure,
+#     and reduce request volume. If an attacker will try accessing the system, this will reduce the risk of brute-force and Denial-of-service to the end users caused by high request rate.
```

Comment:
```suggestion
#     The server allows restricting the limit of unauthenticated requests. It is recommended to turn it on as a security and reliability measure,
#     and to reduce request volume. If an attacker tries accessing the system, this will reduce the risk of brute-force and Denial-of-service to the end users caused by high request rate.
```